### PR TITLE
Set cache_regions=True in the case of s3fs

### PR DESCRIPTION
**Pull request by eapolinario**
# TL;DR
Set cache_regions to circumvent an aiobotocore bug

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 According to https://github.com/fsspec/s3fs/issues/701#issuecomment-1480303225, setting `cache_regions=True` makes it so that `aiobotocore` does not hit the access denied bug.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by pingsutw — APPROVED** [review]
test (windows x python3.11) is failing in all the PRs. I'm fixing it.
